### PR TITLE
Cleaned up KeyTable docs

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: python/pyhail/keytable.py [hunk 1]
```diff
@@ -79,7 +79,13 @@ def nrows(self):
...
 
     def same(self, other):
-        """Test whether two key-tables are identical
+        """Test whether two KeyTables are identical.
+
+        **Examples**
+
+        >>> kt1 = hc.import_keytable("data/kt1.tsv")
+        >>> kt2 = hc.import_keytable("data/kt2.tsv")
+        >>> kt1.same(kt2)
```
Comment (cseed) [inline python/pyhail/keytable.py]: Use the return value of `same`.  Otherwise there's no point.

File: python/pyhail/keytable.py [hunk 2]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
+
+        **Examples**
 
-        :param str output: Output file path
+        Rename column names of KeyTable and export to file:
 
-        :param str types_file: Output path of types file
+        >>> kt = (hc.import_keytable("data/kt1.tsv")
+        >>>         .rename({'column1' : 'newColumn1'}))
+        >>> kt.export("data/kt1_renamed.tsv")
```
Comment (cseed) [inline python/pyhail/keytable.py]: You don't need the intermediate variable `kt`.

File: python/pyhail/keytable.py [hunk 3]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
 
-        :param bool keep: Keep rows where annotation expression evaluates to True
+        **Notes**
+
+        The scope for ``condition`` is all column names in the input :class:`KeyTable`.
+
+        For more information, see the documentation on writing `expressions <../overview.html#expressions>`_
+        and using the `Hail Expression Language <../reference.html#HailExpressionLanguage>`_.
+
+        :param str condition: Annotation expression.
```
Comment (cseed) [inline python/pyhail/keytable.py]: Do you want to include the caution about missing, too?

File: python/pyhail/keytable.py [hunk 4]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
+        :type condition: str or list of str
+
+        :return: A KeyTable with new columns specified by ``condition``.
 
         :rtype: :class:`.KeyTable`
         """
         try:
-            return KeyTable(self.hc, self.jkt.annotate(code))
+            if isinstance(condition, list):
+                condition = ','.join(condition)
```
Comment (cseed) [inline python/pyhail/keytable.py]: I'd put this outside the try.

File: python/pyhail/keytable.py [hunk 5]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
-        and non-overlapping field names.
+        """Join two KeyTables together.
+
+        **Examples**
+
+        Join ``kt1`` to ``kt2`` to produce ``kt3``:
+
+        >>> kt1 = hc.import_keytable("data/kt1.tsv")
+        >>> kt2 = hc.import_keytable("data/kt2.tsv")
+        >>> kt3 = kt1.join(kt2)
```
Comment (cseed) [inline python/pyhail/keytable.py]: Don't need intermediate kt1 and k2.  I'd call the file `example.tsv`.  I don't want to give the impression that KeyTables are a special kind of TSV.

File: python/pyhail/keytable.py [hunk 6]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
+        Hail supports four types of joins specified by ``how``:
+
+         - **inner** -- Key must be present in both ``kt1`` and ``kt2``.
+         - **outer** -- Key present in ``kt1`` or ``kt2``. For keys only in ``kt1``, the value of non-key columns from ``kt2`` is set to missing.
+           Likewise, for keys only in ``kt2``, the value of non-key columns from ``kt1`` is set to missing.
+         - **left** -- Key present in ``kt1``. For keys only in ``kt1``, the value of non-key columns from ``kt2`` is set to missing.
+         - **right** -- Key present in ``kt2``. For keys only in ``kt2``, the value of non-key columns from ``kt1`` is set to missing.
+
+        .. note::
+            Both key-tables must have identical key schemas and non-overlapping column names.
```
Comment (cseed) [inline python/pyhail/keytable.py]: Only the non-keys need be overlapping.

File: python/pyhail/keytable.py [hunk 7]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
 
         :param  right: Key-table to join
         :type right: :class:`.KeyTable`
 
         :param str how: Method for joining two tables together. One of "inner", "outer", "left", "right".
 
+        :return: A KeyTable that is the result of joining two KeyTables as described above.
```
Comment (cseed) [inline python/pyhail/keytable.py]: I don't think you need to say "as described above".

File: python/pyhail/keytable.py [hunk 8]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
-        :param agg_code: Named aggregation expression(s).
-        :type agg_code: str or list of str
+        **Notes**
+
+        The scope for both ``key_condition`` and ``agg_condition`` is all column names in the input :class:`KeyTable`.
+
+        For more information, see the documentation on writing `expressions <../overview.html#expressions>`_
+        and using the `Hail Expression Language <../reference.html#HailExpressionLanguage>`_.
+
+        :param key_condition: Named expression(s) for which columns are keys.
```
Comment (cseed) [inline python/pyhail/keytable.py]: This is the expression to _compute_ the keys of the result.

File: python/pyhail/keytable.py [hunk 9]
```diff
@@ -92,87 +98,202 @@ def same(self, other):
...
     def forall(self, code):
-        """Tests whether a condition is true for all rows
+        """Test whether a condition is true for all rows.
+
+        **Examples**
+
+        Test whether all rows in the KeyTable have the value of ``C1`` equal to 5:
+
+        >>> kt = hc.import_keytable('data/kt1.tsv')
+        >>> kt.exists("C1 == 5")
```
Comment (cseed) [inline python/pyhail/keytable.py]: Again, use return value.  For example,

```
if kt.exists("C1 == 5"):
  print("Got a C1 == 5!")
```

File: python/pyhail/keytable.py [hunk 10]
```diff
@@ -245,67 +375,72 @@ def expand_types(self):
...
-            q: Double
-          }
-          b: Int
-          c: Struct {
-            x: String
-            y: Array[Struct {
-              z: Map[Int]
-            }]
-          }
+        Consider a KeyTable with signature
```
Comment (cseed) [inline python/pyhail/keytable.py]: Can we add an actual example and give this KeyTable a name (kt)?

File: python/pyhail/keytable.py (hunk 5 above)
Comment (jigold) [inline python/pyhail/keytable.py]: I changed the file names, but I think it is better to have the intermediate key tables.

File: python/pyhail/keytable.py (hunk 6 above)
Comment (jigold) [inline python/pyhail/keytable.py]: I don't understand your comment. I think what I have is correct.

File: python/pyhail/keytable.py [hunk 11]
```diff
@@ -79,7 +79,14 @@ def nrows(self):
...
     def same(self, other):
-        """Test whether two key-tables are identical
+        """Test whether two KeyTables are identical.
+
+        **Examples**
+
+        >>> kt1 = hc.import_keytable("data/example1.tsv")
+        >>> kt2 = hc.import_keytable("data/example2.tsv")
+        >>> if kt1.same(kt2):
+        >>>     print "KeyTables are the same!"
```
Comment (cseed) [inline python/pyhail/keytable.py]: We should import print_function from __future__ and use parens:

```# Python 2 and 3:
from __future__ import print_function    # (at top of module)

print('Hello', 'Guido')
```

See: http://python-future.org/compatible_idioms.html

File: python/pyhail/keytable.py [hunk 12]
```diff
@@ -92,87 +99,207 @@ def same(self, other):
...
-        """Export key-table to a TSV file.
+        """Export to a TSV file.
+
+        **Examples**
 
-        :param str output: Output file path
+        Rename column names of KeyTable and export to file:
 
-        :param str types_file: Output path of types file
+        >>> kt = (hc.import_keytable("data/example.tsv")
```
Comment (cseed) [inline python/pyhail/keytable.py]: Don't assign `kt`, export shouldn't return anything.

**Pull request merged**